### ISSUE TITLE
Stacking meta-learner over XGBoost + Skellam + EloMOV (replace simple ensemble)

### ISSUE THREAD

**Issue by lopeztech** [lopeztech/fantasy-coach#171]
## Context

We have three usefully-diverse predictors:
- **XGBoost** (`src/fantasy_coach/models/xgboost_model.py`) — production, nonlinear on the full feature set.
- **Skellam / Poisson margin** (`src/fantasy_coach/models/skellam.py`, #132) — margin-native, coherent win/margin probabilities.
- **EloMOV** (`src/fantasy_coach/models/elo_mov.py`, #126) — fast, low-variance, margin-weighted rating.

Closed #56 (`src/fantasy_coach/models/ensemble.py`) combines them with a simple **average of calibrated probabilities**. That throws away the information that XGBoost should be weighted more heavily in high-data regimes and Elo should be weighted more heavily when a feature is missing. A proper **stacking meta-learner** (a logistic regression trained on out-of-fold base-model predictions) typically adds 0.3–0.8 pp accuracy on top of a plain average, for zero new data and minimal runtime cost.

## Files to edit

- `src/fantasy_coach/models/ensemble.py` — add a `StackedEnsemble` class alongside the existing `AverageEnsemble`. Training interface:
  ```python
  StackedEnsemble.fit(X, y, timestamps, base_models=[xgb, skellam, elo])
  ```
  Under the hood: walk-forward folds (reuse `evaluation/harness.py` fold generator), collect each base model's out-of-fold predicted probabilities, train a logistic-regression meta-learner on those three columns (+ optional 4th column = `missing_odds` flag so the meta-learner can rebalance when market data is absent).
- `src/fantasy_coach/models/loader.py` — extend the artefact-sniffer to dispatch `model_type="stacked"` (matching the pattern set by #84).
- `src/fantasy_coach/__main__.py` — new subcommand `train-stacked --seasons ...` that fits each base model on full data, collects OOF preds, fits the meta-learner, and writes a single joblib blob of shape `{"model_type": "stacked", "feature_names": FEATURE_NAMES, "base_models": [...], "meta": LogisticRegression(...)}`.
- `src/fantasy_coach/predictions.py` — `_compute_contributions` needs a stacked variant: decompose the final log-odds into base-model contributions (3 numbers) for the "why this pick" UI. Keep the per-feature attribution accessible from the XGBoost sub-model for the existing MatchDetail panel.
- `tests/test_models_ensemble.py` — new unit tests: (a) meta-learner respects OOF discipline (no leakage from base model trained on its own fold's validation rows), (b) stacked prediction reproduces each base model at degenerate weight inputs.
- `tests/test_baseline_metrics.py` — add a stacked row to `EXPECTED`; keep XGBoost as production for now, gate promotion on the ablation.
- `docs/model.md` — extend the "Ensembles" section with stacking results; update the "Why not stacking?" note (which preceded #56) if it's still there.

## Details of the change

- **Walk-forward OOF discipline is the whole point** — if you train base models on fold F and meta-learner on fold F's predictions, the meta-learner sees clean OOF predictions exactly as it will at inference time. Sklearn's `StackingClassifier` with `cv=TimeSeriesSplit` is the safe default.
- Meta-learner choice: logistic regression (interpretable, three coefficients tell you "how much weight per base model"). XGBoost as meta is overkill and invites overfitting on 424 rows.
- Calibration layer (#53, Platt/isotonic) applies **after** stacking, not inside each base model, to avoid double-calibration.
- Promotion gate: only swap the production artefact if stacked beats XGBoost on **all three** of accuracy, log-loss, Brier on walk-forward, with the `EXPECTED` dict refreshed in the same PR.

## Why this improves the application

- The three base models already make **genuinely different mistakes** — the three-way consensus panel (#140) visibly shows this. Averaging captures some of that diversity; stacking captures more.
- The meta-learner's coefficients are a free interpretability artefact: if Skellam gets coefficient 0.1 and XGBoost gets 0.8, we learn that Skellam is barely contributing and can save the compute by dropping it.
- Directly de-risks single-model failure modes (e.g., the R8 Tigers/Raiders miss in #165, #166 where XGBoost alone was overconfident against the market).
- Works as a **force multiplier** on everything else in the backlog: any feature we add improves XGBoost, which the stacker then correctly reweights up.

## Dependencies

Best run after #167 (Optuna HPO) so the XGBoost base model is already tuned before stacking measures its residual-correcting value. Not strictly blocked.

## References

- `src/fantasy_coach/models/ensemble.py`, `elo_mov.py`, `skellam.py`, `xgboost_model.py`
- `src/fantasy_coach/models/loader.py` (artefact sniffer, #84)
- `src/fantasy_coach/evaluation/harness.py`
- `docs/model.md` § "Ensembles"
- Closed: #53 (calibration), #56 (simple ensemble), #126, #132, #136.
